buzzmobile/process/gps_mapper/gps_mapper.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python
+
+import rospy
+import math
+import polyline as pl
+import interpolate
+from cv_bridge import CvBridge, CvBridgeError
+from sensor_msgs.msg import NavSatFix
+from sensor_msgs.msg import Image
+from std_msgs.msg import Float64
+from std_msgs.msg import String
+
+class Frames(object):
+    def __init__(self):
+        bearing = None
+        points = None
+        location = None
+
+#Global Variables
+bridge = CvBridge()
+frames = Frames()
+gps_model_pub = rospy.Publisher('gps_model', Image, queue_size=0)
+x_scale = y_scale = 5000
+y_range = x_range = 0
+
+def set_points(polyline):
+    if polyline is not None:
+        points = pl.decode(polyline.data)
+        y_range, x_range, top_left, bottom_right = interpolate.dimensions(frames.points)
+        frames.points = [(y, -x) for (x, y) in points]
+        _, _, top_left, bottom_right = interpolate.dimensions(frames.points)
+        height = y_range * y_scale
+        width = x_range * x_scale
+        frames.points = interpolate.normalized_points(frames.points, int(round(width)), int(round(height)))
+
+def update_image():
+    if hasattr(frames, 'points'):
+        _, _, top_left, bottom_right = interpolate.dimensions(frames.points)
+        height = int(round(y_range * y_scale))
+        width = int(round(x_range * x_scale))
+        point = (frames.location[0], -frames.location[1])
+        if y_range is not 0:
+            point = interpolate.normalize_single_point(y_range, x_range, height, width, top_left, bottom_right, point)
+        result = interpolate.xwindow(frames.points, point, frames.bearing)
+        result_msg = bridge.cv2_to_imgmsg(result, encoding='mono8')
+        gps_model_pub.publish(result_msg)
+
+def set_bearing(angle):
+    frames.bearing = angle.data
+    if hasattr(frames, 'location'):
+        update_image()
+
+def set_location(fix_location):
+    frames.location = (fix_location.latitude, fix_location.longitude)
+    if hasattr(frames, 'bearing'):
```

Comment:
Doesn't this also need `points` to be set? Should we be checking for it here too?